packages/app-mobile/components/getResponsiveValue.ts

```diff
@@ -0,0 +1,73 @@
+// getResponsiveValue() returns the corresponding value for
+// of a particular device screen width based on a valueMap argument
+// and breakpoints
+//
+// Breakpoints:
+// sm: < 481
+// md: >= 481
+// lg: >= 769px
+// xl: >= 1025px
+// xxl: >= 1201px
+//
+// Eg. [ 10, 15, 20, 25, 30 ] means { sm: 10, md: 15, lg: 20, xl: 25, xxl: 30 }
+// [10] and [10, 15] are equivalent to { sm: 10 } and { sm: 10, md: 15 } respectively
+//
+// More Info: https://discourse.joplinapp.org/t/week-4-report/26117
+import { Dimensions } from 'react-native';
+
+export interface ValueMap {
+    sm?: number;
+    md?: number;
+    lg?: number;
+    xl?: number;
+    xxl?: number;
+}
+
+export default function getResponsiveValue(valueMap: ValueMap): number {
+	if (Object.keys(valueMap).length === 0) {
+		throw 'valueMap cannot be an empty object!';
+	}
+
+	const width = Dimensions.get('window').width;
+	let value: number;
+	const { sm, md, lg, xl, xxl } = valueMap;
+
+	// This handles cases where certain values are omitted
+	if (xxl) {
+		value = xxl;
+	}
+
+	if (xl) {
+		value = xl;
+	}
+
+	if (lg) {
+		value = lg;
+	}
+
+	if (md) {
+		value = md;
+	}
+
+	if (sm) {
+		value = sm;
+	}
+
+	if (width >= 481 && !!md) {
```

Comment:
```suggestion
	if (width >= 481 && md != null) {
```

If I run this:
```
$ node
Welcome to Node.js v16.16.0.
Type ".help" for more information.
> !!0
false
> !!1
true
```
As such, if someone sets `md` (or `lg`, `xl`, or `xxl`) to `0`, this conditional won't trigger.

(I'm suggesting `!=` and not `!==` because `md != null` returns `false` if `md` is `undefined`, but `md !== null` returns `true` if `md` is `undefined`).

Reply:
Wasn't sure initially if getResponsiveValue should accept a value of zero, but I think it makes sense that is does. Thank you.